test/opsem/garrays.ll

```diff
@@ -0,0 +1,97 @@
+
+declare i32 @unknown1(...) local_unnamed_addr #0
+
+declare void @verifier.assume(i1)
+
+declare void @verifier.assume.not(i1)
+
+declare void @seahorn.fail()
+
+; Function Attrs: noreturn
+declare void @verifier.error() #1
+
+declare void @seahorn.fn.enter() local_unnamed_addr
+
+; Function Attrs: nounwind
+define i32 @main() local_unnamed_addr #2 {
+entry:
+  tail call void @seahorn.fn.enter() #3
+  %0 = tail call i32 bitcast (i32 (...)* @unknown1 to i32 ()*)() #3
+  %1 = icmp eq i32 %0, 0
+  br i1 %1, label %._crit_edge, label %.lr.ph
+
+.lr.ph:                                           ; preds = %entry
+  br label %2
+
+; <label>:2:                                      ; preds = %.lr.ph, %2
+  %.0.i4 = phi i32 [ 1, %.lr.ph ], [ %10, %2 ]
+  %3 = load i32, i32* getelementptr inbounds ([3 x i32], [3 x i32]* @yyyyyy, i32 0, i32 1), align 4, !tbaa !3
+  %4 = icmp eq i32 %3, 2
+  %5 = select i1 %4, i32 2, i32 1
+  %6 = load i32, i32* getelementptr inbounds ([2 x [2 x i32]], [2 x [2 x i32]]* @zzzzzz, i32 0, i32 1, i32 1), align 4, !tbaa !3
+  %7 = icmp eq i32 %6, 7
+  %8 = select i1 %7, i32 2, i32 1
+  %9 = add nsw i32 %5, %.0.i4
+  %10 = add nsw i32 %9, %8
+  store i32 %9, i32* getelementptr inbounds ([3 x i32], [3 x i32]* @yyyyyy, i32 0, i32 2), align 4, !tbaa !3
+  %11 = add nsw i32 %8, %.0.i4
+  store i32 %11, i32* getelementptr inbounds ([2 x [2 x i32]], [2 x [2 x i32]]* @zzzzzz, i32 0, i32 1, i32 0), align 4, !tbaa !3
+  %12 = add nuw nsw i32 %8, %5
+  store i32 %12, i32* getelementptr inbounds ([2 x [2 x i32]], [2 x [2 x i32]]* @zzzzzz, i32 0, i32 0, i32 1), align 4, !tbaa !3
+  %13 = add nsw i32 %.0.i4, 2
+  %14 = trunc i32 %13 to i8
+  store i8 %14, i8* getelementptr inbounds ([4 x i8], [4 x i8]* @.str, i32 0, i32 2), align 1, !tbaa !7
+  %15 = tail call i32 bitcast (i32 (...)* @unknown1 to i32 ()*)() #3
+  %16 = icmp eq i32 %15, 0
+  br i1 %16, label %._crit_edge.loopexit, label %2
+
+._crit_edge.loopexit:                             ; preds = %2
+  br label %._crit_edge
+
+._crit_edge:                                      ; preds = %._crit_edge.loopexit, %entry
+  %17 = load i32, i32* getelementptr inbounds ([3 x i32], [3 x i32]* @yyyyyy, i32 0, i32 2), align 4, !tbaa !3
+  %18 = icmp sgt i32 %17, 0
+  %19 = load i32, i32* getelementptr inbounds ([2 x [2 x i32]], [2 x [2 x i32]]* @zzzzzz, i32 0, i32 0, i32 1), align 4
+  %20 = icmp sgt i32 %19, 0
+  %or.cond = and i1 %18, %20
+  %21 = load i32, i32* getelementptr inbounds ([2 x [2 x i32]], [2 x [2 x i32]]* @zzzzzz, i32 0, i32 1, i32 0), align 4
+  %22 = icmp sgt i32 %21, 0
+  %or.cond3 = and i1 %or.cond, %22
+  br i1 %or.cond3, label %23, label %verifier.error
+
+; <label>:23:                                     ; preds = %._crit_edge
+  tail call void @verifier.assume.not(i1 true) #3
```

Comment:
this looks wrong. The assumption got compiled to `true`, so the test will return `unsat` independently of all other instructions.